cmake/Modules/OpmFind.cmake

```diff
@@ -147,36 +149,58 @@ macro (find_and_append_package_to prefix name)
     if (NOT DEFINED ${name}_FOUND AND NOT DEFINED ${NAME}_FOUND
         OR ("${name}" STREQUAL "opm-common" AND NOT _opm_common_deps_processed))
       string(REGEX MATCH "(dune|opm)-.*" _is_opm ${name})
-      if (${name}_DIR AND NOT (_${name}_exempted OR _is_opm))
-        message (STATUS "Finding package ${name} using config mode ${_${name}_exempted}")
-        find_package (${name} ${ARGN} NO_MODULE PATHS ${${name}_DIR} NO_DEFAULT_PATH)
-      else ()
-        # print message if this neither an opm module nor ecl
-        if (NOT (${name}_exempted EQUAL -1 OR _is_opm_) )
-          message (STATUS "Finding package ${name} using module mode")
+      if(NOT _is_opm)
+        string(REGEX MATCH "ewoms" _is_opm ${name})
+      endif()
+      if(${name} STREQUAL "ecl")
+        # Give us a chance to find ecl installed to CMAKE_INSTALL_PREFIX.
+        # We need to deactivate the package registry for this.
+        create_module_dir_var(ecl)
+        set(ARGN_NO_REQUIRED ${ARGN})
+        if(ARGN)
+          list(REMOVE_ITEM ARGN_NO_REQUIRED "REQUIRED")
         endif()
-		if(${name} STREQUAL "ecl")
-			# Give us a chance to find ecl installed to CMAKE_INSTALL_PREFIX.
-			# We need to deactivate the package registry for this.
-			create_module_dir_var(ecl)
-			set(ARGN_NO_REQUIRED ${ARGN})
-			list(REMOVE_ITEM ARGN_NO_REQUIRED "REQUIRED")
- 			find_package (${name} ${ARGN_NO_REQUIRED} NO_CMAKE_SYSTEM_PACKAGE_REGISTRY NO_CMAKE_PACKAGE_REGISTRY)
-			# If everything else failed fall back to the registry
-		endif()	
-		find_package (${name} ${ARGN})
-	  endif ()
-	endif ()
-	if (NOT DEFINED ${name}_FOUND)
-	  set (${name}_FOUND "${${NAME}_FOUND}")
-	endif ()
-	if (NOT DEFINED ${NAME}_FOUND)
-	  set (${NAME}_FOUND "${${name}_FOUND}")
-	endif ()
-
-       if(name STREQUAL "opm-common")
-         set(_opm_common_deps_processed ON)
-       endif()
+        find_package (${name} ${ARGN_NO_REQUIRED} NO_CMAKE_SYSTEM_PACKAGE_REGISTRY NO_CMAKE_PACKAGE_REGISTRY)
+        # If everything else failed fall back to the registry
```

Comment:
Is this comment coorect? Seems like we just get some properties ecl-config.cmake defines for the target ecl and use them as variables.

Reply:
nope, it should be removed. it clawed itself stuck and refused to go during reorganizations it seems.